core/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -413,7 +413,17 @@ protected PreparedStatement readSnapshotData(Connection connection, String ident
      * @throws SQLException when an exception occurs while creating the event data
      */
     protected DomainEventData<?> getDomainEventData(ResultSet resultSet) throws SQLException {
-        return (DomainEventData<?>) getTrackedEventData(resultSet, null);
+        long globalSequence = resultSet.getLong(schema.globalIndexColumn());
+        TrackingToken trackingToken = new GlobalSequenceTrackingToken(globalSequence);
```

Comment:
As you can see below these methods are never combined, so a `ClassCastException` won't happen.

Call hierarchy for `protected DomainEventData<?> getTrackedEventData(ResultSet resultSet)`
```
BatchingEventStorageEngine.readEventData(TrackingToken, boolean)
    AbstractEventStorageEngine.readEvents(TrackingToken, boolean)
        EventConsumer in EmbeddedEventStore.peekPrivateStream(int, TimeUnit)
        EventProducer in EmbeddedEventStore.fetchData()
        SequenceEventStorageEngine.readEvents(TrackingToken, boolean)
```
Call hierarchy for `protected DomainEventData<?> getDomainEventData(ResultSet resultSet)`
```
BatchingEventStorageEngine.readEventData(String, long)
    AbstractEventStorageEngine.readEvents(String, long)
        AbstractEventStore.readEvents(String, long)
        EventStorageEngine.readEvents(String)
        AbstractEventStore.readEvents(String)
        SequenceEventStorageEngine.readEvents(String, long)
```